server/src/main/java/org/opensearch/index/similarity/SimilarityProviders.java

```diff
@@ -271,13 +273,21 @@ static void assertSettingsIsSubsetOf(String type, Version version, Settings sett
         }
     }
 
-    public static LegacyBM25Similarity createBM25Similarity(Settings settings, Version indexCreatedVersion) {
+    public static Similarity createBM25Similarity(Settings settings, Version indexCreatedVersion) {
         assertSettingsIsSubsetOf("BM25", indexCreatedVersion, settings, "k1", "b", DISCOUNT_OVERLAPS);
 
         float k1 = settings.getAsFloat("k1", 1.2f);
         float b = settings.getAsFloat("b", 0.75f);
         boolean discountOverlaps = settings.getAsBoolean(DISCOUNT_OVERLAPS, true);
 
+        return new BM25Similarity(k1, b, discountOverlaps);
+    }
+
+    public static Similarity createLegacyBM25Similarity(Settings settings, Version indexCreatedVersion) {
```

Comment:
Do we need to support `LegacyBM25Similarity` at all?

Reply:
I see the comment `@deprecated {@link BM25Similarity} should be used instead`, but since it was not announced I thought we should continue ti support the `LegacyBM25Similarity`, but I'm open for thoughts since we are targeting this to 3.0.0 (merge only to `main`) we can remove the `LegacyBM25Similarity`.

Reply:
So how the legacy BM25Similarity is going to be used (in other words, who is the caller of `createLegacyBM25Similarity`)?